firestore/integration_test_internal/src/document_change_test.cc

```diff
@@ -78,5 +78,67 @@ TEST_F(DocumentChangeTest, TestDocumentChanges) {
 
 #endif  // defined(__ANDROID__)
 
+TEST_F(DocumentChangeTest, Equality) {
+  DocumentChange invalid_change_1 = DocumentChange();
+  DocumentChange invalid_change_2 = DocumentChange();
+
+  EXPECT_TRUE(invalid_change_1 == invalid_change_2);
+  EXPECT_FALSE(invalid_change_1 != invalid_change_2);
+
+  CollectionReference collection = Collection();
+  Query query = collection.OrderBy("a");
+
+  DocumentReference doc1 = collection.Document("1");
+  DocumentReference doc2 = collection.Document("2");
+
+  TestEventListener<QuerySnapshot> listener("TestDocumentChanges");
+  ListenerRegistration registration = listener.AttachTo(&query);
+  Await(listener);
+  QuerySnapshot snapshot = listener.last_result();
+  EXPECT_EQ(snapshot.size(), 0);
+
+  WriteDocument(doc1, MapFieldValue{{"a", FieldValue::Integer(1)}});
+  Await(listener);
+  snapshot = listener.last_result();
+
+  std::vector<DocumentChange> changes = snapshot.DocumentChanges();
+  EXPECT_EQ(changes.size(), 1);
```

Comment:
nit: Use `ASSERT_EQ` instead of `EXPECT_EQ` here. By using `EXPECT_EQ`, if the assertion fails then the test will continue. And if the assertion failed because the actual size of the vector was zero then accessing element 0 on line 107 will crash. By using `ASSERT_EQ` the test will be aborted and avoid the undefined behavior of accessing a non-existent element of a vector.

Also on lines 118 and 132.

Reply:
+1.